### PR TITLE
htmlsitemap and seocore added with coding standard

### PR DESCRIPTION
**Pull request by priyankavasoya**
@harishmk1805 
please check

### CODE REVIEW

File: app/code/Dss/HtmlSiteMap/Helper/Data.php [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    public function getCategoryDisable($storeId = null): string
+    {
+        if ($storeId !== null) {
+            $categoryDisable = $this->scopeConfig->getValue(
+                'dss_htmlsitemap/category/id_category',
+                $this->scopeStore,
+                $storeId
+            );
+        } else {
+            $categoryDisable = $this->scopeConfig->getValue('dss_htmlsitemap/category/id_category', $this->scopeStore);
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Helper/Data.php:267]: if used multiple times use as variable

File: app/code/Dss/HtmlSiteMap/Helper/Data.php [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+    }
+
+    /**
+     * Is EnableCategory
+     *
+     * @return bool
+     */
+    public function isEnableCategory(): bool
+    {
+        $enableCategory = $this->scopeConfig->isSetFlag('dss_htmlsitemap/category/enable_category', $this->scopeStore);
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Helper/Data.php:281]: if used multiple times use as variable

File: app/code/Dss/HtmlSiteMap/Helper/Data.php [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+    }
+
+    /**
+     * Is EnableProduct
+     *
+     * @return mixed|bool
+     */
+    public function isEnableProduct(): bool
+    {
+        $enableProduct = $this->scopeConfig->isSetFlag('dss_htmlsitemap/product/enable_product', $this->scopeStore);
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Helper/Data.php:292]: if used multiple times use as variable

File: app/code/Dss/HtmlSiteMap/Model/Config/Source/Order.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+class Order implements \Magento\Framework\Option\ArrayInterface
+{
+    /**
+     * To Option array
+     *
+     * @return array
+     */
+    public function toOptionArray(): array
+    {
+
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Model/Config/Source/Order.php:30]: remove space

File: app/code/Dss/HtmlSiteMap/Observer/AfterCategorySave.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Framework\App\Cache\TypeListInterface as CacheTypeListInterface;
+use Magento\Config\Model\ResourceModel\Config;
+use Dss\HtmlSiteMap\Helper\Data;
+use Dss\SeoCore\Helper\Data as SeoCoreHelper;
+
+class AfterCategorySave implements ObserverInterface
+{
+
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Observer/AfterCategorySave.php:31]: remove unused space

File: app/code/Dss/HtmlSiteMap/Setup/Patch/Data/SitemapProduct.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+            $eavSetup->addAttribute(Product::ENTITY, 'excluded_html_sitemap', [
+                'group' => 'Search Engine Optimization',
+                'type' => 'text',
+                'backend' => \Magento\Eav\Model\Entity\Attribute\Backend\ArrayBackend::class,
+                'frontend' => '',
+                'label' => 'Excluded from HTML Sitemap',
+                'input' => 'select',
+                'class' => '',
+                'source' => \Dss\HtmlSiteMap\Model\Config\Product\ExcludeHtmlSiteMap::class,
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/Setup/Patch/Data/SitemapProduct.php:64]: use Dss\HtmlSiteMap\Model\Config\Product\ExcludeHtmlSiteMap;

File: app/code/Dss/HtmlSiteMap/etc/acl.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<!-- /**
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/etc/acl.xml:2]: ????

File: app/code/Dss/HtmlSiteMap/etc/adminhtml/system.xml [hunk 8]
```diff
@@ -0,0 +1,185 @@
+<?xml version="1.0"?>
+<!-- /**
+ * Digit Software Solutions..
+ *
+ * NOTICE OF LICENSE
+ *
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/etc/adminhtml/system.xml:6]: remove commet from xml

File: app/code/Dss/HtmlSiteMap/view/frontend/layout/sitemap_product_list.xml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * that is bundled with this package in the file LICENSE.txt.
+ *
+ * @category   Dss
+ * @package    Dss_HtmlSiteMap
+ * @author     Extension Team
+ * @copyright Copyright (c) 2024 Digit Software Solutions. ( https://digitsoftsol.com )
+ */ -->
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+    <update handle="sitemap_index_index" />
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/layout/sitemap_product_list.xml:17]: why used update?

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+?>
+<?php
+    $helper = $block->getHelper();
+    $additionUrl = $helper->getAdditionUrl();
+    $titleAddition = $helper->titleAddition();
+    $openNewTab = $helper->openNewTab();
+    $additionUrlCount = $additionUrl;
+
+    $additionLink = $block->getAdditionLink();
+if ($additionUrlCount != null): ?>
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml:31]: please formate phtml use <?php if(): ?>

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        foreach ($additionLink as $value => $link): ?>
+            <?php
+            if ($value % 2 == 1): ?>
+                    <div class="col-dss-3"><h3 class="dss_product_list">
+                    <a href="<?= /* @noEscape */ $link; ?>"
+                <?php
+            else:
+                ?> title="<?= /* @noEscape */ $link; ?>"
+                <?php
+                if ($openNewTab == 1) {
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml:44]: formate phtml

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/custom.phtml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+$baseUrl = $helper->getBaseUrl();
+$getEnable = $helper->isEnable();
+$urlKey = $helper->getModuleRoute();
+$titleSiteMap = $helper->getTitleSiteMap();
+$showLinkAt = $helper->getShowLinkAt();
+?>
+<?php if ($getEnable && $showLinkAt !== 'top'): ?>
+    <li class="nav item">
+        <a href="<?= /* @noEscape */ $baseUrl . $urlKey; ?>">
+            <?= /* @noEscape */  $escaper->escapeHtml($titleSiteMap); ?>
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/custom.phtml:34]: why used /* @noEscape */ ??

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/custom_top.phtml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+$baseUrl = $helper->getBaseUrl();
+$getEnable = $helper->isEnable();
+$urlKey = $helper->getModuleRoute();
+$titleSiteMap = $helper->getTitleSiteMap();
+$showLinkAt = $helper->getShowLinkAt();
+?>
+<?php if ($getEnable && $showLinkAt === 'top'): ?>
+    <li>
+        <a href="<?= /* @noEscape */  $baseUrl . $urlKey; ?>">
+            <?= /* @noEscape */  $escaper->escapeHtml($titleSiteMap); ?>
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/custom_top.phtml:34]: why used /* @noEscape */

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/product_list.phtml [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+?>
+<?php
+$helper = $block->getHelper();
+
+//Product Link
+$enableProduct = $helper->isEnableProduct();
+$titleProduct = $helper->titleProduct();
+
+if ($enableProduct): ?>
+    <h2 class="dss_h2"><?= /* @noEscape */ $titleProduct; ?></h2>
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/product_list.phtml:31]: remove /* @noEscape */

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+$helper = $block->getHelper();
+
+//Store View Link
+$additionUrl = $helper->getAdditionUrl();
+$baseUrl = $helper->getBaseUrl();
+$enableStoreView = $helper->isEnableStoreView();
+$titleStore = $helper->titleStore();
+?>
+<?php
+if ($enableStoreView): ?>
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml:33]: formate phtml as per coding standards

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+            <div class="col-dss-3">
+                <h3 class="dss_product_list view-<?= $escaper->escapeHtml($lang->getCode()) ?>">
+                    <a href="<?= $escaper->escapeUrl($block->getTargetStoreRedirectUrl($lang)) ?>">
+                        <?= $escaper->escapeHtml($lang->getName()) ?>
+                    </a>
+                </h3>
+            </div>
+        <?php endforeach; ?>
+
+        <?php
```
Comment (harishmk1805) [inline app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml:58]: <?php endforeach; ?>
    </div>
    <?php endif; ?>

File: app/code/Dss/SeoCore/Block/Validate.php [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Validate constructor.
+     *
+     * @param Data $dataHelper
+     * @param Context $context
+     * @param array $data
+     */
+    public function __construct(
+        private Data $dataHelper,
+        protected Context $context,
```
Comment (harishmk1805) [inline app/code/Dss/SeoCore/Block/Validate.php:35]: remove protected it's required?

File: app/code/Dss/SeoCore/Helper/Data.php [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        return $this->storeManager->getStore()->getCode();
+    }
+
+    /**
+     * Get store manager
+     *
+     * @return \Magento\Store\Model\StoreManagerInterface
+     */
+    public function getStoreManager(): \Magento\Store\Model\StoreManagerInterface
```
Comment (harishmk1805) [inline app/code/Dss/SeoCore/Helper/Data.php:71]: use Magento\Store\Model\StoreManagerInterface;

File: app/code/Dss/SeoCore/etc/adminhtml/system.xml [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+ * @category   Dss
+ * @package    Dss_SeoCore
+ * @author     Extension Team
+ * @copyright Copyright (c) 2024 Digit Software Solutions. ( https://digitsoftsol.com )
+ */ -->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
+    <system>
+        <!-- <tab id="dss_seo_suite" translate="label" sortOrder="300" class="dss_commerce_seo">
+            <label><![CDATA[Dss Commerce SEO]]></label>
```
Comment (harishmk1805) [inline app/code/Dss/SeoCore/etc/adminhtml/system.xml:19]: remove

**Review by harishmk1805 — CHANGES_REQUESTED** [review]
please check phtml file proper

File: app/code/Dss/HtmlSiteMap/Block/CategoryCollection.php [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+        return $this->categoryHelper;
+    }
+
+    /**
+     * Get Cms page
+     *
+     * @return \Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getCmsPages(): \Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection
```
[the remark this answers is not in the document]
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Block/CategoryCollection.php]: Done

File: app/code/Dss/HtmlSiteMap/Block/ItemsCollection.php [hunk 21]
```diff
@@ -0,0 +1,383 @@
...
+    {
+        return $this->_storeManager->getStore()->getId();
+    }
+
+    /**
+     * Get Helper
+     *
+     * @return \Dss\HtmlSiteMap\Helper\Data
+     */
+    public function getHelper(): \Dss\HtmlSiteMap\Helper\Data
```
[the remark this answers is not in the document]
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Block/ItemsCollection.php]: done

File: app/code/Dss/HtmlSiteMap/Block/ItemsCollection.php [hunk 22]
```diff
@@ -0,0 +1,383 @@
...
+     */
+    public function __construct(
+        protected Context $context,
+        public Data $helper,
+        protected PostHelper $postDataHelper,
+        protected EncoderInterface $encoder,
+        protected UrlHelper $urlHelper,
+        array $data = []
+    ) {
+        $this->scopeConfig = $context->getScopeConfig();
```
[the remark this answers is not in the document]
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Block/ItemsCollection.php]: done

File: app/code/Dss/HtmlSiteMap/Block/ProductCollection.php [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * This source file is subject to the EULA
+ * that is bundled with this package in the file LICENSE.txt.
+ *
+ * @category   Dss
+ * @package    Dss_HtmlSiteMap
+ * @author     Extension Team
+ * @copyright Copyright (c) 2024 Digit Software Solutions. ( https://digitsoftsol.com )
+ */
+
```
[the remark this answers is not in the document]
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Block/ProductCollection.php:18]: done

File: app/code/Dss/HtmlSiteMap/Controller/Index/Index.php [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+     * Index constructor.
+     *
+     * @param Context $context
+     * @param PageFactory $resultPageFactory
+     * @param RequestInterface $request
+     * @param Data $helper
+     * @param UrlInterface $url
+     */
+    public function __construct(
+        protected Context $context,
```
[the remark this answers is not in the document]
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Controller/Index/Index.php]: done

File: app/code/Dss/HtmlSiteMap/Helper/Data.php (hunk 1 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Helper/Data.php:281]: done

File: app/code/Dss/HtmlSiteMap/Helper/Data.php (hunk 3 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Helper/Data.php:306]: done

File: app/code/Dss/HtmlSiteMap/Model/Config/Source/Order.php (hunk 4 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Model/Config/Source/Order.php]: done

File: app/code/Dss/HtmlSiteMap/Observer/AfterCategorySave.php (hunk 5 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Observer/AfterCategorySave.php]: done

File: app/code/Dss/HtmlSiteMap/Setup/Patch/Data/SitemapProduct.php (hunk 6 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/Setup/Patch/Data/SitemapProduct.php]: done

File: app/code/Dss/HtmlSiteMap/etc/acl.xml (hunk 7 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/etc/acl.xml]: done

File: app/code/Dss/HtmlSiteMap/etc/adminhtml/system.xml (hunk 8 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/etc/adminhtml/system.xml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/layout/sitemap_product_list.xml (hunk 9 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/layout/sitemap_product_list.xml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml (hunk 10 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml (hunk 11 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/additional_list.phtml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/custom.phtml (hunk 12 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/custom.phtml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/custom_top.phtml (hunk 13 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/custom_top.phtml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/product_list.phtml (hunk 14 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/product_list.phtml]: done

File: app/code/Dss/SeoCore/etc/adminhtml/system.xml (hunk 19 above)
Comment (priyankavasoya) [reply app/code/Dss/SeoCore/etc/adminhtml/system.xml]: done

File: app/code/Dss/SeoCore/Helper/Data.php (hunk 18 above)
Comment (priyankavasoya) [reply app/code/Dss/SeoCore/Helper/Data.php]: done

File: app/code/Dss/SeoCore/Block/Validate.php (hunk 17 above)
Comment (priyankavasoya) [reply app/code/Dss/SeoCore/Block/Validate.php]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml (hunk 16 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml]: done

File: app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml (hunk 15 above)
Comment (priyankavasoya) [reply app/code/Dss/HtmlSiteMap/view/frontend/templates/store_list.phtml:33]: done

**Review by harishmk1805 — APPROVED** [review]

**Pull request merged**